src/pathwayvis/services/ws.ts

```diff
@@ -69,10 +69,8 @@ export class WSService {
     localWs.onmessage = (event) => {
       const result = JSON.parse(event.data);
       const requestId = result['request-id'];
-      const callback = _.find(this._callbacks, 'id', requestId);
-      console.log("ON MESSAGE");
+      const callback = this._callbacks.find((obj) => obj.id === requestId);
```

Comment:
I know you just removed the `_` calls from this function, but thinking about it we could make it a bit more robust.
If there's no callback with the requestId, `callback` will be undefined and in https://github.com/DD-DeCaF/metabolica-ui-map/pull/138/files/a11d3b2a3c072c1d59f23f2e98fed92173fc694a..f03790e147e96d5b61260d3677e09e910d452282#diff-d3a93e6dfa8a3ead607f0b6b58ac6ab8R74 it will result in an error. Could you please add a guard?

```
if (!callback) {
  throw new Error(`No callback was found for request ${requestId}`)
}
```